### PR TITLE
Fix rabbit mq bindings in manifest.

### LINKED ISSUE #1973: The RabbitMQ container uses the wrong binding in the connection string
When generating the manifest for RabbitMQ containers it generates the following:

```JSON
{
  "resources": {
    "rabbit": {
      "type": "container.v0",
      "image": "rabbitmq:3-management",
      "env": {
        "RABBITMQ_DEFAULT_USER": "guest",
        "RABBITMQ_DEFAULT_PASS": "{rabbit.inputs.password}"
      },
      "bindings": {
        "tcp": {
          "scheme": "tcp",
          "protocol": "tcp",
          "transport": "tcp",
          "containerPort": 5672
        },
        "management": {
          "scheme": "http",
          "protocol": "tcp",
          "transport": "http",
          "containerPort": 15672
        }
      },
      "connectionString": "amqp://guest:{rabbit.inputs.password}@{rabbit.bindings.management.host}:{rabbit.bindings.management.port}",
      "inputs": {
        "password": {
          "type": "string",
          "secret": true,
          "default": {
            "generate": {
              "minLength": 10
            }
          }
        }
      }
    }
  }
}
```

Which is mostly correct except that the connection string uses the wrong binding (the management one).

### PR DESCRIPTION
**Pull request by mitchdenny**
Fixes https://github.com/dotnet/aspire/issues/1973

... and then some. I change `AppHost` over to using Rabbit MQ container since we don't yet have `rabbitmq.v0` support in AZD. This allows us to deploy eshoplite via AZD. As part of that work I had to move the EF data context that was in CatalogDb app into a shared library so it could be shared by CatalogService. Having CatalogService referencing CatalogDb created some publish time issues and this was the easiest workaround.

Whilst the app now deploys cleanly to AZD there are some issues associated with scaling. Specifically the rabbit mq container scales to zero. I am not 100% sure but I don't think you can scale up when you aren't using a HTTP binding in ACA.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1985)

### CODE REVIEW

File: src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs [hunk 1]
```diff
@@ -26,7 +26,6 @@ public static IResourceBuilder<RabbitMQContainerResource> AddRabbitMQContainer(t
         var rabbitMq = new RabbitMQContainerResource(name, password);
         return builder.AddResource(rabbitMq)
                        .WithAnnotation(new EndpointAnnotation(ProtocolType.Tcp, port: port, containerPort: 5672))
-                       .WithAnnotation(new EndpointAnnotation(ProtocolType.Tcp, uriScheme: "http", name: "management", port: null, containerPort: 15672))
```
Comment (davidfowl) [inline src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs:29]: No more management UI?

File: playground/eShopLite/AppHost/aspire-manifest.json [hunk 2]
```diff
@@ -32,7 +32,38 @@
...
+          "protocol": "tcp",
+          "transport": "tcp",
+          "containerPort": 5672
+        },
+        "management": {
+          "scheme": "http",
+          "protocol": "tcp",
+          "transport": "http",
+          "containerPort": 15672
+        }
```
Comment (davidfowl) [inline playground/eShopLite/AppHost/aspire-manifest.json:53]: Why does this still show up?

Comment (mitchdenny) [reply playground/eShopLite/AppHost/aspire-manifest.json:53]: Hrm. Maybe I didn't check in the updated version. I'll fix it up

File: src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs (hunk 1 above)
Comment (mitchdenny) [reply src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs:29]: No. I think we need to move it to an extension anyway (I think this is something we discussed).

I was also looking at changing the container image but there isn't a generic 3.x tag that doesn't include the management stuff.

We need to have a think about the way we handle container image version defaults in aspire.

Comment (davidfowl) [reply src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs:29]: File an issue and change the image to the one that doesn't have the UI

Comment (mitchdenny) [reply src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs:29]: I ended up spotting the `"3"` tagged image. I just missed as I was scanning through the set of tags. I'll still file a work item for discussing how we want to handle container image version management.

Comment (mitchdenny) [reply src/Aspire.Hosting/RabbitMQ/RabbitMQBuilderExtensions.cs:29]: https://github.com/dotnet/aspire/issues/1997

**Review by davidfowl — APPROVED** [review]

**Pull request merged**